File: AquariumControllerESPIDF/AquariumController/main/Settings/Alarm.cpp
#include "Alarm.h"

Alarm::Alarm() {
  this->name = "";
  this->alarmState = 0;
  this->setAlarmStateFunction = NULL;
}
void Alarm::init(String name, String shortName, SetAlarmStateFunction setAlarmStateFunction) {
  this->name = name;
  this->shortName = shortName;
  String alarmKey= this->shortName + "_K";
  this->alarmState = savedState.getULong(alarmKey.c_str(), 0);
  this->setAlarmStateFunction = setAlarmStateFunction;
}
String Alarm::getName() {
  return this->name;
}
unsigned long Alarm::getAlarmState() {
  return this->alarmState;
}
void Alarm::setAlarmState(unsigned long newState) {
  this->alarmState = newState;
  String alarmKey= this->shortName + "_K";
  savedState.putULong(alarmKey.c_str(), newState);
  this->setAlarmStateFunction(this);
}

File: AquariumControllerClient/AquariumControllerClient/Alarm.swift
//
//  Alarm.swift
//  AquariumControllerClient
//
//  Created by Connor Williams on 7/7/24.
//

import Foundation
import SwiftUI

class Alarm: HashableClass, Identifiable, ObservableObject {
    var id: String
    var name: String
    @Published var alarmState: Int
    @Published var alarmOverride: Bool
    init(_ name: String) {
        self.id = name
        self.name = name
        self.alarmState = 0
        self.alarmOverride = false
    }
    func getName() -> String {
        return self.name
    }
    func getAlarmState() -> Int {
        return self.alarmState
    }
    func setAlarmState(newState: Int) {
        self.alarmState = newState
    }
    func getAlarmOverride() -> Bool {
        return self.alarmOverride
    }
    func setAlarmOverride(override: Bool) {
        self.alarmOverride = override
    }
}
